src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -26,13 +26,20 @@ namespace OpenTelemetry.Metrics
     internal sealed class AggregatorStore
     {
         private static readonly ObjectArrayEqualityComparer ObjectArrayComparer = new ObjectArrayEqualityComparer();
+        private static readonly StringArrayEqualityComparer StringArrayComparer = new StringArrayEqualityComparer();
         private readonly object lockZeroTags = new object();
         private readonly HashSet<string> tagKeysInteresting;
         private readonly int tagsKeysInterestingCount;
 
+        private readonly ConcurrentDictionary<string[], string[]> tagKeyCombinations =
+            new ConcurrentDictionary<string[], string[]>(StringArrayComparer);
+
+        private readonly ConcurrentDictionary<object[], object[]> tagValueCombinations =
```

Comment:
This crossed my mind too, but I considered something like

```
counter.Add(10, new("A",1), new("C",3), new("B",2));
counter.Add(10, new("D",1), new("E",3), new("F",2));
```

I think this would actually work fine as is, but it does seem a little goofy